chore: add script to update the index used by the omnibar

akash-codemonk opened the pull request:
Related to https://github.com/appsmithorg/appsmith/issues/17501

Adds a script to update the index used by the omnibar.

The script fetches all the markdown under docs/ folder and parses, uploads the same. It updates the href of links in the markdown to full paths. The images for the omnibar are fetched from `https://raw.githubusercontent.com/appsmithorg/appsmith-docs/main/website/static`. The title is the h1 text in the markdown. 

Also updates the workflow to run the script.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/appsmithorg/appsmith-docs?pullRequest=564) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/appsmithorg/appsmith-docs?pullRequest=564) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/appsmithorg/appsmith-docs?pullRequest=564) it.</sub>

akash-codemonk commented:
> I see that we are deleting the `package-lock.json` file. Is this intentional and we don't need it? Can you elaborate please? If not, can we restore it please?

Ah no idea. Just did npm install and it probably did the changes. I will revert that back.

sharat87 reviewed (changes requested):
Can you run `npm install` on this branch once? Since we added two dependencies to `package.json`, the `package-lock.json` should include corresponding changes as well. But I don't see any non-whitespace changes in that file here.

Just run `npm install` once, and then commit the `package-lock.json` file as is. Then we'll be able to use `npm ci` instead of `npm i` in the workflow.

akash-codemonk commented:
> Can you run `npm install` on this branch once? Since we added two dependencies to `package.json`, the `package-lock.json` should include corresponding changes as well. But I don't see any non-whitespace changes in that file here.
> 
> Just run `npm install` once, and then commit the `package-lock.json` file as is. Then we'll be able to use `npm ci` instead of `npm i` in the workflow.

Updated
